hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java

```diff
@@ -911,4 +925,103 @@ public boolean isMaster(Address address) {
         Address masterAddress = clusterService.getMasterAddress();
         return server.doAddressesMatch(masterAddress, address);
     }
+
+    public void printSecurityInfo() {
+        securityLogger.info(String.format(
+                "Enable DEBUG/FINE log level for log category %1$s "
+                        + " or use -D%1$s system property to see 🔒 security recommendations and the status of current config.",
+                SECURITY_BANNER_CATEGORY));
+        boolean showInfoSecurityBanner = System.getProperty(SECURITY_BANNER_CATEGORY) != null;
+        if ((showInfoSecurityBanner && securityLogger.isInfoEnabled()) || securityLogger.isFineEnabled()) {
+            printSecurityFeaturesInfo(getConfig(), showInfoSecurityBanner ? Level.INFO : Level.FINE);
+        }
+    }
+
+    @SuppressWarnings({ "checkstyle:CyclomaticComplexity", "checkstyle:MethodLength" })
+    private void printSecurityFeaturesInfo(Config config, Level logLevel) {
+        StringBuilder sb = new StringBuilder("\n🔒Security recommendations and their status:");
+        addSecurityFeatureCheck(sb, "Use a custom cluster name", !Config.DEFAULT_CLUSTER_NAME.equals(config.getClusterName()));
+        boolean isMulticastJoin = shouldUseMulticastJoiner(getActiveMemberNetworkConfig(config).getJoin());
+        addSecurityFeatureCheck(sb, "Disable member multicast discovery/join method", !isMulticastJoin);
+
+        AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
+        addSecurityFeatureCheck(sb, "Use advanced networking, separate client and member sockets",
+                advancedNetworkConfig.isEnabled());
+        boolean bindAny = getProperties().getBoolean(SOCKET_SERVER_BIND_ANY);
+        addSecurityFeatureCheck(sb,
+                "Bind Server sockets to a single network interface (disable " + SOCKET_SERVER_BIND_ANY.getName() + ")",
+                !bindAny);
+        StringBuilder tlsSb = new StringBuilder();
+        boolean tlsUsed = true;
+        if (advancedNetworkConfig.isEnabled()) {
+            for (Map.Entry<EndpointQualifier, EndpointConfig> e : advancedNetworkConfig.getEndpointConfigs().entrySet()) {
+                tlsUsed = addAdvNetworkTlsInfo(tlsSb, e.getKey(), e.getValue().getSSLConfig()) && tlsUsed;
+            }
+        } else {
+            SSLConfig sslConfig = config.getNetworkConfig().getSSLConfig();
+            tlsUsed = addSecurityFeatureCheck(tlsSb, "Use TLS communication protection (Enterprise)",
+                    sslConfig != null && sslConfig.isEnabled());
+        }
+        boolean jetEnabled = config.getJetConfig().isEnabled();
+        if (jetEnabled) {
+            boolean trustedEnv = tlsUsed || !bindAny;
+            addSecurityFeatureCheck(sb, "Use Jet in trusted environments only (single network interface and/or TLS enabled)",
```

Comment:
Using `single network interface` is actually not enough, right? You need to bind to localhost.

For the other part - is tls enough, don't we need authentication as well?

Reply:
A trusted network doesn't need to be the localhost only. Members should be aware of such a requirement and they should come up with the proper environment.

With TLS it's again about awareness. I don't want to introduce too complex checks (e.g. `TLS.mutual==REQUIRED || (TLS && Hazelcast.security.authentication used)`).